src/utils/Cypher.ts

```diff
@@ -26,19 +35,176 @@ export default class Cypher {
     return this.configuration;
   }
 
-  public static encrypt(data: string): string {
+  public static getCrypto(): CryptoSetting {
+    if (!this.cryptoSetting) {
+      this.cryptoSetting = {
+        key: this.configuration.key
+      };
+      throw new BackendError({
+        source: Constants.CENTRAL_SERVER,
+        module: MODULE_NAME,
+        method: 'getCrypto',
+        message: 'Crypto Setting is missing'
+      });
+    }
+    return this.cryptoSetting;
+  }
+
+  public static async detectConfigurationKey(tenantID: string): Promise<boolean> {
+    let isKeyChanged = false;
+    const configCryptoKey: string = Configuration.getCryptoConfig().key;
+    const keySettings = await SettingStorage.getCryptoSettings(tenantID);
+    let cryptoSettingToSave: CryptoSetting;
+    // TODO check for migrationDone
+
+    // Check if Crypto Settings exist
```

Comment:
Initial crypto settings shall be added at tenant creation. 
A default crypto settings configuration setup code must be added and called at tenant creation and here for safety. 
A random secure key shall be generated.